pkg/chunk/aws_storage_client.go

```diff
@@ -435,12 +442,36 @@ func (a awsStorageClient) GetChunks(ctx context.Context, chunks []Chunk) ([]Chun
 		return nil, err
 	}
 
-	dynamoDBChunks, err = a.getDynamoDBChunks(ctx, dynamoDBChunks)
+	gangSize := a.cfg.DynamoDBChunkGangSize * dynamoDBMaxReadBatchSize
+	if gangSize == 0 { // zero means turn feature off
+		gangSize = len(dynamoDBChunks)
+	}
+
+	results := make(chan chunksPlusError)
```

Comment:
We have a few other implementations of "parallel map a function that returns `(a, err)`". Some of the others use a buffered channel. I don't think it's necessary here, because the loop immediately following will drain all of `results` as quickly as possible. 

No action required, just flagging in case my reading is wrong and you notice something on a second glance.

Reply:
Yes, a buffer would be extra memory management overhead for no obvious benefit.